CardanoSharp.Wallet/Extensions/Models/NativeScriptExtension.cs

```diff
@@ -15,11 +15,34 @@ public static byte[] GetPolicyId(this NativeScript nativeScript)
             return HashUtility.Blake2b244(buffer.ToArray());
         }
 
+        public static CBORObject GetCBOR2(this NativeScript nativeScript)
+        {
+            if(nativeScript.ScriptPubKey != null)
+                return nativeScript.ScriptPubKey.GetCBOR();
+
+            if (nativeScript.ScriptAll != null)
+                return nativeScript.ScriptAll.GetCBOR();
+
+            if (nativeScript.ScriptAny != null)
+                return nativeScript.ScriptAny.GetCBOR();
+
+            if (nativeScript.ScriptNofK != null)
+                return nativeScript.ScriptNofK.GetCBOR();
+
+            if (nativeScript.InvalidAfter != null)
+                return nativeScript.InvalidAfter.GetCBOR();
+
+            if (nativeScript.InvalidBefore != null)
+                return nativeScript.InvalidBefore.GetCBOR();
+
+            return null;
+        }
```

Comment:
@Daniel-SchaeferJ @tweakch @safestak-keith 
thoughts on making this better. GetCBOR() and GetCBOR2() don't exactly seem like the best approach